google-ads-examples/src/main/java/com/google/ads/googleads/examples/planning/GenerateHistoricalMetrics.java

```diff
@@ -0,0 +1,157 @@
+
+    @Parameter(
+        names = ArgumentNames.CUSTOMER_ID,
+        description = "The customer in which to create a new keyword plan.")
+    public Long customerId;
+
+    @Parameter(
+        names = ArgumentNames.KEYWORD_PLAN_ID,
+        description = "The keyword plan ID for which to generate metrics.")
+    public Long keywordPlanId;
+  }
+
+  public static void main(String[] args) {
+    GenerateHistoricalMetricsParams params = new GenerateHistoricalMetricsParams();
+    if (!params.parseArguments(args)) {
+
+      // Optional, specify the customer ID for which the call is made.
+      params.customerId = Long.valueOf("INSERT_CUSTOMER_ID");
+
+      params.keywordPlanId = Long.valueOf("INSERT_KEYWORD_PLAN_ID");
+    }
+    GoogleAdsClient googleAdsClient = null;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      System.exit(1);
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      System.exit(1);
+    }
+
+    try {
+      new GenerateHistoricalMetrics()
+          .runExample(googleAdsClient, params.customerId, params.keywordPlanId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Runs the code example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param planId the plan ID.
+   */
+  // [START generate_historical_metrics]
+  private void runExample(GoogleAdsClient googleAdsClient, Long customerId, Long planId) {
+    String resourceName = ResourceNames.keywordPlan(customerId, planId);
+
+    try (KeywordPlanServiceClient keywordPlanServiceClient =
+        googleAdsClient.getLatestVersion().createKeywordPlanServiceClient()) {
+      GenerateHistoricalMetricsResponse response =
+          keywordPlanServiceClient.generateHistoricalMetrics(resourceName);
+
+      for (KeywordPlanKeywordHistoricalMetrics metric : response.getMetricsList()) {
+        // These metrics include those for both the search query and any close variants included in
+        // the response.
+
+        ProtocolStringList variants = metric.getCloseVariantsList();
```

Comment:
Can we just declare this as `List<String>`? I think the `ProtocolStringList` type could be distracting, and AFAICT we're not doing anything here that requires methods from that specific subclass of `List<String>`.